### DISCUSSION TITLE
Know Micro Frontends? Say hi!

### DISCUSSION THREAD

**Discussion by webcrumbs-community** [webcrumbs-community/webcrumbs#116] [General]
Attention, WebCrumbs micro frontend virtuosos!

This is your hub for collaboration:

- Jump into the discussion with a comment, or
- Catch a mention from your code comrades.

We're curating a stream of fresh issues and vibrant dialogues tailored for your specialized expertise. 
Stay connected, stay informed!

**Comment by webcrumbs-community** [comment]
Here comes to my mind @aditya4196 and @rovoskarl, but I'd love to hear other names too!

**Reply by Ashok-Kumar-dharanikota** [reply]
🙋‍♂️ can I join too ?,  not an expert but like to learn, join and contribute. 😀

**Reply by webcrumbs-community** [reply]
Sure, @Ashok-Kumar-dharanikota. Thank you! All help is welcome with our core issues. I'd recommend playing with `core/experiment-1` and then reading about our next challenges (issues). You'll also find useful materials at `research/resources`.

**Comment by dew-dr0p** [comment]
Happy to be on this side of the team, I've been working on the module federation experiment, and I've been able to make some progress though it's taking a while, cos I'm just getting introduced to micro-frontends, but I always jump at opportunities to learn something new.
I will be away for a while (around 3 weeks), so I might not be able to actively finish what I've started concerning the module federation. If it's done, before I'm back great then, but meanwhile, I'll continue to carry out more research especially on how to implement our end goal, beginning from this module federation standpoint. I should be able to get something definite, by the time I'm back. And if the issue is still open, I'd love to drop whatever ideas I've got.

Once again I'd like to say, It's been an interesting journey thus far @webcrumbs-community, and I'm enjoying every bit of it.  Love this community.

**Reply by webcrumbs-community** [reply]
That's the spirit, @dew-dr0p! We are all learners. Happy you're with us!

You may like to check out this [resources on micro frontends](https://github.com/webcrumbs-community/webcrumbs/blob/main/research/Resources.md) we've pulled together.

Enjoy your time off! Take care.

**Reply by dew-dr0p** [reply]
Great to mention I'm back and would love to contribute more to this community and see us move forward

**Reply by dew-dr0p** [reply]
Great resource you've pulled together for micro frontends. Any discount code for the udemy course? Would love to purchase it @webcrumbs-community .

**Reply by webcrumbs-community** [reply]
Yes! Here's the coupon I have for this month: https://www.udemy.com/course/microfrontend-course/?couponCode=36AA4D338B4-DISCORD

**Comment by webcrumbs-community** [comment]
Hey, there! @dew-dr0p @Ashok-Kumar-dharanikota @vbrazo @rovoskarl 
Just added a new issue #155. I changed it to "first come first served" for us to move faster!
If you are interested in the challenge, feel free to try to solve it and sending a PR.

**Comment by Ashok-Kumar-dharanikota** [comment]
Hi @webcrumbs-community ,

Thank You for asking.  

I'm planning to dive back into the project this week. There are some concepts, like Micro-frontends and creating plugins, that I want to get more comfortable with. It'll definitely set me up for the long run.

Big shoutout and good luck 👍 to all the awesome contributors tackling those issues. I'll be jumping in real soon.

**Reply by aditya4196** [reply]
Hi Ashok!!
Im in the same state with micro-frontends. 
We can work on it together. 
Let me know when you have time availabilities this week to connect to discuss about it.

**Reply by Ashok-Kumar-dharanikota** [reply]
Sure! 😀 I will let you know tomorrow if it's possible.

**Reply by Ashok-Kumar-dharanikota** [reply]
Hello @aditya4196,

I  recently explored the codebase and ran experiment-1. I noticed that plugin-1 and plugin-2 run on separate server ports and are seamlessly imported into localhost:3000.

I have a question regarding the real deployment of the project. If we have n number of plugins, will each one run on a separate port? I'm curious to know what the end user and developers will achieve after deployment. Could you please provide an example to help me understand this better? Your thoughts on this matter would be greatly appreciated. It helps me to see the bigger picture.

**Reply by aditya4196** [reply]
Actually when it comes to production, I was thinking it to be more like the containers will spun up for all those plugins when the request for that plugin will be made by site.
Then using the concept of port forwarding to access the plugins
But still in thought process. @webcrumbs-community can explain it better if im not correct about the process

**Reply by webcrumbs-community** [reply]
Hey, guys! Exactly. In production plugins will be js files served from one unique url and different endpoints, e.g. something like registry.webcrumbs.org/plugin1/latest, registry.webcrumbs.org/plugin2/v1 , etc. Experiments end up being small steps towards our goal. Make sense?

**Reply by aditya4196** [reply]
Yes!!
So those js plugin files would be be exposed as instances unique to the site, right?

**Reply by webcrumbs-community** [reply]
Yes! Like an "npm" for all webcrumbs' plugins
